website/src/main/java/com/google/step/data/OrganizationInfo.java

```diff
@@ -0,0 +1,61 @@
+package com.google.step.data;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import java.util.Objects;
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import org.apache.commons.validator.UrlValidator;
+
+public final class OrganizationInfo {
+  private final Entity entity;
+
+  public OrganizationInfo(HttpServletRequest request) {
+    this.entity = getOrgEntityFrom(request);
+  }
+
+  private Entity getOrgEntityFrom(HttpServletRequest request) {
+    Entity newOrganization = new Entity("Organization");
+    newOrganization.setProperty("name", request.getParameter("orgName"));
+    newOrganization.setProperty("webLink", request.getParameter("webLink"));
+    newOrganization.setProperty("donateLink", request.getParameter("donateLink"));
+    newOrganization.setProperty("about", request.getParameter("about"));
+    newOrganization.setProperty("flags", new ArrayList<String>());
+
+    return newOrganization;
+  }
+
+  public Boolean isValid() {
+    UrlValidator urlValidator = new UrlValidator();
+    //Required fields
+    if (((String) this.entity.getProperty("name")).isEmpty() || 
+    	  ((String) this.entity.getProperty("name")).isEmpty() ||
+        !urlValidator.isValid(((String) this.entity.getProperty("webLink")))) {
+      return false;
+    }
+    //Change invalid Optional Fields to null
+    if (urlValidator.isValid(((String) this.entity.getProperty("donateLink")))) {
+      this.entity.setProperty("donateLink",null);
+    }
+    return true;
+  }
+
+  public void compare(Entity duplicate) {
```

Comment:
I strongly suggest renaming this.

Usually "compare" describes a function which returns a negative number, 0, or a positive number, and does not perform any mutations.

I think a more descriptive name for this would be "merge".